main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Dialogs/ConfigureRepositoryDialog.cs

```diff
@@ -0,0 +1,220 @@
+				BackgroundColor = Ide.Gui.Styles.BackgroundColor
+			};
+
+			separator = new HSeparator ();
+
+			repository = new Label (GettextCatalog.GetString ("Repository") + ":") {
+				Margin = new WidgetSpacing (0, 6, 0, 6)
+			};
+
+			moduleContainer = new HBox ();
+
+			moduleName = new Label (GettextCatalog.GetString ("Module name") + ":") {
+				WidthRequest = 80
+			};
+
+			entryName = new TextEntry ();
+			messageContainer = new HBox ();
+
+			message = new Label (GettextCatalog.GetString ("Message") + ":") {
+				WidthRequest = 80
+			};
+
+			entryMessage = new TextEntry ();
+
+			buttonContainer = new HBox {
+				Margin = new WidgetSpacing (0, 6, 0, 0)
+			};
+
+			cancelButton = new Button (GettextCatalog.GetString ("Cancel"));
+			okButton = new Button (GettextCatalog.GetString ("Configure"));
+			currentEditor = repo.VersionControlSystem.CreateRepositoryEditor (repo);
+
+		}
+
+		void BuildGui ()
+		{
+			Title = GettextCatalog.GetString ("Configure Repository");
+
+			Width = 500;
+
+			contentContainer.PackStart (currentEditor.Widget, true, true);
+			currentEditor.Show ();
+
+			repoContainer.PackStart (contentContainer, true);
+			repoContainer.PackStart (separator);
+			repoContainer.PackStart (repository);
+
+			moduleContainer.PackStart (moduleName);
+			moduleContainer.PackStart (entryName, true);
+
+			messageContainer.PackStart (message);
+			messageContainer.PackStart (entryMessage, true);
+
+			repoContainer.PackStart (moduleContainer);
+			repoContainer.PackStart (messageContainer);
+
+			entryName.Sensitive = hasChanges;
+			entryMessage.Sensitive = hasChanges;
+
+			buttonContainer.PackEnd (okButton);
+			buttonContainer.PackEnd (cancelButton);
+
+			repoContainer.PackEnd (buttonContainer);
+
+			Content = repoContainer;
+			Resizable = false;
+		}
+
+		void AttachEvents ()
+		{
+			UrlBasedRepositoryEditor edit = currentEditor as UrlBasedRepositoryEditor;
+			if (edit != null) {
+				edit.UrlChanged += OnEditUrlChanged;
+				edit.PathChanged += OnEditUrlChanged;
+			}
+
+			if (hasChanges) {
+				entryMessage.Changed += OnMessageChanged;
+			}
+
+			cancelButton.Clicked += OnCancel;
+			okButton.Clicked += OnModify;
+		}
+
+		protected override void Dispose (bool disposing)
+		{
+			UrlBasedRepositoryEditor edit = currentEditor as UrlBasedRepositoryEditor;
+			if (edit != null) {
+				edit.UrlChanged -= OnEditUrlChanged;
+				edit.PathChanged -= OnEditUrlChanged;
+			}
+
+			if (hasChanges) {
+				entryMessage.Changed -= OnMessageChanged;
+			}
+
+			cancelButton.Clicked -= OnCancel;
+			okButton.Clicked -= OnModify;
+
+			base.Dispose (disposing);
+		}
+
+		void OnMessageChanged (object sender, EventArgs e)
+		{
+			okButton.Sensitive = !string.IsNullOrEmpty (entryMessage.Text);
+		}
+
+		void OnEditUrlChanged (object sender, EventArgs e)
+		{
+			UpdateCheckoutButton ();
+		}
+
+		void OnCancel (object sender, EventArgs e)
+		{
+			Respond (Command.Cancel);
+		}
+
+		void OnModify (object sender, EventArgs e)
```

Comment:
OnModify does not actually show the intent of the handler:

```suggestion
		void OnOkClicked (object sender, EventArgs e)
```

Reply:
Changed!